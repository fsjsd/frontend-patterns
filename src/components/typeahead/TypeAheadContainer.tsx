import React, { Suspense } from 'react'
import { ContentContainer } from '../../ux/ContentContainer'
import FieldSet from '../../ux/designsystem/FieldSet'
import InputText from '../../ux/designsystem/InputText'
import Label from '../../ux/designsystem/Label'
import Legend from '../../ux/designsystem/Legend'
import Loading from '../../ux/Loading'
import DemoError from './DemoError'
import DemoUniversitySearch from './DemoUniversitySearch'

/**
 * Container for the Typeahead component
 * @returns JSX
 */
const TypeAheadContainer = () => {

  return (
    <ContentContainer
      title='Type-ahead'
      codeLink="/features/typeahead"
      markDownPromise={import('./requirements.md')}
    >
      <Suspense fallback={<Loading />}>
        <div role="main">
          <DemoUniversitySearch />
          <DemoError />
          <FieldSet>
            <Label htmlFor="error">Normal text box</Label>
            <InputText placeholder='Press tab' />
            <Legend>Just on screen to test tab key press</Legend>
          </FieldSet>
        </div>
      </Suspense>
    </ContentContainer>
  )
}

export default TypeAheadContainer