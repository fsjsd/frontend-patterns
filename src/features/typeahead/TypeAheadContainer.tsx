import React, { Suspense } from 'react'
import ContentHeader, { ContentHeaderLabel } from '../../ux/ContentHeader'
import { ContentWrapper } from '../../ux/ContentWrapper'
import Loading from '../../ux/Loading'
const TypeAhead = React.lazy(() => import('./TypeAhead'))

const TypeAheadContainer = () => {
  return (<>
    <ContentHeader>
      <ContentHeaderLabel>Type-ahead</ContentHeaderLabel>
    </ContentHeader>
    <ContentWrapper>
      <Suspense fallback={<Loading />}>
        <div role="main">
          <TypeAhead />
        </div>
      </Suspense>
    </ContentWrapper>
  </>
  )
}

export default TypeAheadContainer