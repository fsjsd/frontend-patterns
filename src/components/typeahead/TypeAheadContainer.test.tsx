import React from 'react'
import TypeAheadContainer from "./TypeAheadContainer";
import { render, waitFor } from '@testing-library/react'

describe("TypeAhead", () => {
  it("renders and matches snapshot", async () => {
    const { container, queryByRole } = render(<TypeAheadContainer />)
    await waitFor(() => expect(queryByRole("main")).toBeInTheDocument());
    expect(container).toMatchSnapshot();
  })
})