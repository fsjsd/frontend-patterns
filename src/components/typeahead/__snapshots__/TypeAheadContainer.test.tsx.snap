// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypeAhead renders and matches snapshot 1`] = `
.c1 {
  font-size: 1em;
  font-weight: inherit;
  color: inherit;
  padding: 13px;
  -webkit-box-flex: 2;
  -webkit-flex-grow: 2;
  -ms-flex-positive: 2;
  flex-grow: 2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1.25em;
  padding: 0;
  display: none;
}

.c2 a {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
  line-height: normal;
  color: rgb(0 100 176);
  -webkit-text-decoration: underline;
  text-decoration: underline;
  -webkit-transition: opacity 0.2s ease-in-out;
  transition: opacity 0.2s ease-in-out;
}

.c3 {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 15px 15px 12px 15px;
  border-bottom: solid 3px transparent;
  border-bottom: solid 3px rgb(0,127,224);
  opacity: 0.6;
}

.c3:hover {
  color: rgb(0,145,255);
  opacity: 1;
}

.c4 {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 15px 15px 12px 15px;
  border-bottom: solid 3px transparent;
  opacity: 0.6;
}

.c4:hover {
  color: rgb(0,145,255);
  opacity: 1;
}

.c0 {
  background-color: white;
  border-bottom: solid 1px #e0e0e0;
  box-shadow: 0 -6px 15px 0px rgb(0 0 0 / 10%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
}

.c5 {
  padding: 15px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  overflow-y: auto;
  -webkit-scroll-behavior: smooth;
  -moz-scroll-behavior: smooth;
  -ms-scroll-behavior: smooth;
  scroll-behavior: smooth;
  -webkit-overflow-scrolling: touch;
}

.c6 {
  padding: 0 0 1rem 0;
}

.c9 {
  color: inherit;
  font-size: 0.95rem;
  font-family: inherit;
  line-height: 1rem;
  margin-top: 0.25rem;
  display: block;
  width: -moz-fit-content;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  border-radius: 0.375rem;
  border-color: rgb(209 213 219 / 1);
  box-shadow: 0 0 #0000,0 1px 2px 0 rgb(0 0 0 / 0.05);
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: #fff;
  border-width: 1px;
  padding-top: 0.5rem;
  padding-right: 0.75rem;
  padding-bottom: 0.5rem;
  padding-left: 0.75rem;
  margin: 0;
}

.c9:focus {
  border-color: rgb(99 102 241 / 1);
  outline: 2px solid transparent;
  outline-offset: 2px;
  box-shadow: rgb(255,255,255) 0px 0px 0px 0px,rgb(99,102,241) 0px 0px 0px 1px,rgba(0,0,0,0.05) 0px 1px 2px 0px;
}

.c7 {
  padding: 4px 0px;
  display: block;
  font-size: 0.95rem;
  line-height: 1rem;
  font-weight: 500;
  color: rgb(55 65 81 / 1);
}

.c10 {
  display: block;
  float: left;
  padding: 4px 0px;
  font-size: 0.75rem;
  line-height: 1rem;
  font-weight: 400;
  color: rgb(55 65 81 / 1);
}

.c8 {
  box-sizing: border-box;
}

@media (min-width:768px) {
  .c2 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

<div>
  <header
    class="c0"
  >
    <h2
      class="c1"
    >
      Type-ahead
    </h2>
    <div
      class="c2"
    >
      <div
        class="c3"
        role="button"
        title="Run demo"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M10 16.5l6-4.5-6-4.5zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
          />
        </svg>
      </div>
      <div
        class="c4"
        role="button"
        title="View notes"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M3 18h12v-2H3v2zM3 6v2h18V6H3zm0 7h18v-2H3v2z"
          />
        </svg>
      </div>
      <a
        aria-label="View source code on GitHub"
        href="https://github.com/fsjsd/frontend-patterns/tree/main/src/features/typeahead"
        rel="noopener noreferrer"
        role="button"
        target="_blank"
      >
        <div
          class="c4"
          title="View code on github"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.013 8.013 0 0 0 16 8c0-4.42-3.58-8-8-8z"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </a>
    </div>
  </header>
  <section
    class="c5"
    role="document"
  >
    <div
      role="main"
    >
      <fieldset
        class="c6"
      >
        <label
          class="c7"
          for="university-search"
        >
          Search Australian Universities
        </label>
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-owns="university-search-list"
          class="c8"
          role="combobox"
        >
          <input
            aria-autocomplete="list"
            aria-controls="typeahead-list"
            class="c9"
            id="university-search"
            placeholder="Search..."
            role="textbox"
            value=""
          />
        </div>
        <legend
          class="c10"
        >
          Run code on local to bypass hipolabs.com CORS policy
        </legend>
      </fieldset>
      <fieldset
        class="c6"
      >
        <label
          class="c7"
          for="error"
        >
          Error demo
        </label>
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-owns="error-list"
          class="c8"
          role="combobox"
        >
          <input
            aria-autocomplete="list"
            aria-controls="typeahead-list"
            class="c9"
            id="error"
            role="textbox"
            value=""
          />
        </div>
        <legend
          class="c10"
        >
          Demos graceful failure of suggestions if getResults fails
        </legend>
      </fieldset>
      <fieldset
        class="c6"
      >
        <label
          class="c7"
          for="error"
        >
          Normal text box
        </label>
        <input
          class="c9"
          placeholder="Press tab"
        />
        <legend
          class="c10"
        >
          Just on screen to test tab key press
        </legend>
      </fieldset>
    </div>
  </section>
</div>
`;
